### DISCUSSION TITLE
WSockets connection limit?

### DISCUSSION THREAD

**Discussion by jeffery30162** [uNetworking/uWebSockets.js#672] [Q&A]
Hi, Is there a connection limit to uWebSockets?

I am using hyper-express for uWebSockets and am currently seeing a limit of 15345 connections. is this a hard cap for uWebSockets?
How can I increase this limit per single instance?

Thanks

**Comment by ghost** [comment]
Your OS has limits, lookup ulimit -n and so on.

**Comment by jeffery30162** [comment]
With a single and multiple instances of WSockets.js on the same ubnutu machine I am only able to reach a maximum of around 22k connections. Any idea on how to increase this?

**Reply by ghost** [reply]
https://alexhultman.medium.com/millions-of-active-websockets-with-node-js-7dc575746a01

**Reply by jeffery30162** [reply]
Looking at the client and server examples used within that demo, it looks like there are multiple ip addresses being used for the server. Does that mean per client 20k connections would need to connect a different ip address on the server? How does this work in the real world? A vps with 50 public ips?

**Reply by ghost** [reply]
You can do the opposite instead; 1 server IP and many client IPs.

**Reply by jeffery30162** [reply]
How does that work?

**Reply by ghost** [reply]
The same exact way. You should look this up yourself and learn from it. You specify what IP to use when connecting just like you would specify what IP to listen to, in the opposite

**Reply by e3dio** [reply]
> How does this work in the real world? A vps with 50 public ips?

@jeffery30162  I think you are very confused, "real world" clients connect with their own IP address, you only need 1 IP for your server, the 1 million example is creating many  IP because there is only 1 computer making 1 million connections to 1 other computer, TCP connections need 1 IP/Port per connection. You did not give any details about how you are testing this, you are probably running some benchmark on localhost and using up all available ports

**Reply by jeffery30162** [reply]
> > How does this work in the real world? A vps with 50 public ips?
> 
> @jeffery30162 I think you are very confused, "real world" clients connect with their own IP address, you only need 1 IP for your server, the 1 million example is creating many IP because there is only 1 computer making 1 million connections to 1 other computer, TCP connections need 1 IP/Port per connection. You did not give any details about how you are testing this, you are probably running some benchmark on localhost and using up all available ports

I understand that, but there seems to be a limit per network interface around 20k concurrent connections.

**Reply by ghost** [reply]
Your network interface knows nothing about what a connection is, there is no limit per interface. There is a 20k limit per IP in your particular configuration. You need more IP addresses. Please read a book about this.
